psiminer-core/src/main/kotlin/psi/language/JavaHandler.kt

```diff
@@ -14,8 +14,13 @@ class JavaHandler : LanguageHandler() {
     override val classPsiType = PsiClass::class.java
     override val methodPsiType = PsiMethod::class.java
 
-    override fun collectFunctionCallsIdentifiers(root: PsiElement): List<PsiElement> =
-        PsiTreeUtil
-            .collectElementsOfType(root, PsiMethodCallExpression::class.java)
-            .map { it.methodExpression.lastChild }
+    override fun actionOnRecursiveCall(root: PsiElement, action: (PsiElement) -> Unit) =
+        PsiTreeUtil.collectElementsOfType(root, PsiMethodCallExpression::class.java)
+            .filter {
+                val resolvedMethod = it.resolveMethod()
+                PsiManager.getInstance(root.project).areElementsEquivalent(root, resolvedMethod)
+            }
+            .forEach {
+                it.methodExpression.lastChild.apply(action)
```

Comment:
The fact that the action applies to the very last child only is a bit confusing given the current function name